src/client/Microsoft.Identity.Client/Platforms/net6/KeyMaterialManager.cs

```diff
@@ -0,0 +1,270 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Security.Cryptography;
+using System.Security.Cryptography.X509Certificates;
+using System.Threading;
+using Microsoft.Identity.Client.Core;
+using Microsoft.Identity.Client.Credential;
+using Microsoft.Identity.Client.PlatformsCommon.Interfaces;
+using Microsoft.Identity.Client.PlatformsCommon.Shared;
+
+namespace Microsoft.Identity.Client.Platforms.netcore
+{
+    /// <summary>
+    /// Class to store crypto key information for a Managed Identity supported Azure resource.
+    /// For more details see https://aka.ms/msal-net-managed-identity
+    /// </summary>
+    internal class KeyMaterialManager : IKeyMaterialManager
+    {
+        private const string IsKeyGuardEnabledProperty = "Virtual Iso";
+        private CryptoKeyType _cryptoKeyType = CryptoKeyType.None;
+        private const string KeyProviderName = "Microsoft Software Key Storage Provider";
+        private const string MachineKeyName = "ManagedIdentityCredentialKey";
+        private const string SoftwareKeyName = "ResourceBindingKey";
+        private readonly X509Certificate2 _bindingCertificate;
+        private readonly CertificateCache _certificateCache;
+        private readonly object _keyInfoLock = new(); // Lock object
+        private readonly ILoggerAdapter _logger;
+
+        public KeyMaterialManager(ILoggerAdapter logger)
+        {
+            _logger = logger;
+
+            _certificateCache = CertificateCache.Instance();
+            
+            _bindingCertificate = _certificateCache.GetOrAddCertificate(
```

Comment:
I don't think this is correct. The lifetime of this object is the same as the lifetime of the ManagedIdentityApplication, which can be the same as the lifetime of the process. 

I think you should check if the cert needs to be rotated on every request. So instead of having a property `BidingCertificate` you have a method `GetOrCreateBindingCertificate`  which does the rotation?

Reply:
(this will also eliminate the need to leak cert expiration outside this class)